apps/lockfile-explorer-web/src/parsing/lfxGraphSerializer.ts

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import type { IJsonLfxDependency, IJsonLfxEntry, IJsonLfxGraph } from './JsonLfxGraph';
+import { LfxGraph, LockfileDependency, LockfileEntry } from './LfxGraph';
+
+export function serializeToJson(graph: LfxGraph): IJsonLfxGraph {
+  const jsonLfxEntries: IJsonLfxEntry[] = [];
+
+  const jsonIdByEntry: Map<LockfileEntry, number> = new Map();
+
+  function toJsonId(entry: LockfileEntry): number {
+    const result: number | undefined = jsonIdByEntry.get(entry);
+    if (result === undefined) {
+      throw new Error('Attempt to serialize disconnected object');
+    }
+    return result;
+  }
+
+  for (const entry of graph.entries) {
+    const nextIndex: number = jsonLfxEntries.length;
+
+    const jsonLfxEntry: IJsonLfxEntry = {
+      jsonId: nextIndex,
+
+      kind: entry.kind,
+      entryId: entry.entryId,
+      rawEntryId: entry.rawEntryId,
+      packageJsonFolderPath: entry.packageJsonFolderPath,
+      entryPackageName: entry.entryPackageName,
+      displayText: entry.displayText,
+      entryPackageVersion: entry.entryPackageVersion,
+      entrySuffix: entry.entrySuffix,
+
+      // Lists will be added in the second loop
+      dependencies: [],
+      transitivePeerDependencies: [],
+      referrerJsonIds: []
+    };
+
+    jsonLfxEntries.push(jsonLfxEntry);
+    jsonIdByEntry.set(entry, jsonLfxEntry.jsonId);
+  }
+
+  // Now that we built the jsonId lookup, we can serialize the lists.
+  for (let i: number = 0; i < jsonLfxEntries.length; ++i) {
+    const jsonLfxEntry: IJsonLfxEntry = jsonLfxEntries[i];
+    const entry: LockfileEntry = graph.entries[i];
+
+    for (const dependency of entry.dependencies) {
+      const jsonLfxDependency: IJsonLfxDependency = {
+        name: dependency.name,
+        version: dependency.version,
+        entryId: dependency.entryId,
+        dependencyType: dependency.dependencyType,
+        peerDependencyMeta: {
+          name: dependency.peerDependencyMeta.name,
+          version: dependency.peerDependencyMeta.version,
+          optional: dependency.peerDependencyMeta.optional
+        }
```

Comment:
```suggestion
        ...dependency,
        peerDependencyMeta: {
          ...dependency.peerDependencyMeta
        }
```

Reply:
Same point as before.  `...` is not appropriate in a serializer.